### ISSUE TITLE
Default ride name

### ISSUE THREAD

**Issue by runelaenen** [OpenRCT2/OpenRCT2#2086]
Hello everyone,

I know this really is a tiny feature to add, but one that solves something that bothered me since I started playing RCT years ago..
The first of any give ride you build is always named "<Ride name> 1"

Would it be possible to remove the number, and only start using this system if more than one of a specific ride is built?

Currently:
Placing down the first ferris wheel -> Name is "Ferris wheel 1"
Placing down the second ferris wheel -> Name is "Ferris wheel 2"

By adding this feature:
Placing down the first ferris wheel -> Name is "Ferris wheel"
Placing down the second first ferris wheel -> Name is "Ferris wheel 2" 
(an addition would be to only add the "2" if the first ferris wheel is still named "Ferris wheel" ;)

Original forum post: https://openrct.net/forum/topic/599/last/default-ride-name

**Comment by HaasJona** [comment]
I honestly like the 1 at the end more. Assume you have 3 Ferris Wheels, which message is more intuitive:

"Ferris Wheel 1 broke down"
"Ferris Wheel broke down"

For me it's the first one because it makes clear which one broke down (the first one you build). You might be aware that the other Ferris Wheels would be labelled 2 and 3 but it's not intuitive.

Also these ugly default names force you more or less to rename them anyway which is a good thing in my opinion.

**Comment by duncanspumpkin** [comment]
It can always just be an option. Although I think we might start to have far too many options at some point.

**Comment by IntelOrca** [comment]
@HaasJona Its not going to be a problem, we can automatically change the non-numbered ride back to having a 1 if you build a second. And it will be an option to do it anyway.

**Comment by Gymnasiast** [comment]
An option for something this minor is silly. As you probably know, I think OpenRCT2 already contains too many options for minor things, and they complicate the code base far more than I feel it's worth. I think we should either implement it the way IntelOrca describes (adding a 1 when you build a second Ferris Wheel) or not at all. After all, it takes a few seconds to chop off the 1 if you don't like it.

**Comment by IntelOrca** [comment]
@Gymnasiast IMO we have more trivial options than this... And we aren't anywhere close to the number of options OpenTTD has.

**Comment by Gymnasiast** [comment]
@IntelOrca : Yes, we have more trivial options than this. That doesn't justify adding another, and I would also be in favour of pruning some of the existing ones.

**Comment by IntelOrca** [comment]
@Gymnasiast Its inevitable that we are going to have a lot of options. We should try to make a great interface that handles that well.

**Comment by Gymnasiast** [comment]
@IntelOrca : I'm sure we'll gain many more options, but I am against making every little thing an option. This one is way too trivial. We should either implement it or not.
Also, an interface that handles a lot of options well does not solve the added complexity in the code for every feature, let alone the amount of configurations that have to be supported and tested.

**Comment by daihakken** [comment]
As a user like me, I would prefer developers to finish it later if you are really want to. (maybe 1.0+) I think it would spent a lot of time to set up a logic (?) system to handle that thing, but in return the function is insignificant for average joe like me.

**Comment by IntelOrca** [comment]
@Gymnasiast I disagree with this particular option, I think it should exist.

**Comment by pizza2004** [comment]
Are we really having another fight over more options?  As a user I have to say the main thing I liked about OpenTTD was the options.  It really allowed me to configure the game just exactly the way I wanted for my own enjoyment.  I'd love it if it checked to see if there was a ride named "Whatever 1" or "Whatever" and always just built the new ride with the lowest number that isn't already used.

**Comment by ccfreak2k** [comment]
OpenTTD does have a lot of options, but it keeps most of them in a separate window. As long as the current behavior is the default, I don't think it'll be an issue.

**Comment by e-foley** [comment]
I tend to side with Gymnasiast: I don't think this is significant enough to warrant an option.

**Comment by JarnoVgr** [comment]
Maybe it's not significant enough for an option, but this seems like a viable solution to me:
"Its not going to be a problem, we can automatically change the non-numbered ride back to having a 1 if you build a second."

**Comment by IntelOrca** [comment]
There is also possibility of a hidden option... an option that is in the config INI but not on the options window. Like window snap radius.

**Comment by Gymnasiast** [comment]
I still think it's way too trivial to warrant an option and the resulting added complexity, but I'm not going to be that hard-headed. I can agree with having a config INI option (on by default).

@pizza2004 : More options may be nice for you, but many users have trouble finding their way trough loads of options, and they complicate the code base, and also increase the number of configurations to support and the chance of bugs. That is why I am opposed to trivial options. I think they're not worth the hassle.

**Comment by IntelOrca** [comment]
@Gymnasiast There will have to be an advanced mode then which shows more options when enabled.

**Comment by pizza2004** [comment]
@Gymnasiast To be clear, I think this is trivial and isn't worth an option, but at the time there were a lot of issues where people were talking about adding options and it just turned into this exact discussion over whether or not we should have a lot of options. :D

I like the idea that it automatically renames it to add the 1 when you build a second one.  It would be the best solution in this case I think.

**Comment by gdabi** [comment]
I found very interesting the way OpenTTD manages such a large number of options. It has a "main" options window, which displays the most important options, and a very simple "registry"-like window which lets you configure a lot of trivial things.

I always like to include a lot of options in my software, but I think that decompiling and refactoring the source code must be a priority (mainly because decompiling generates a very poorly quality software, which is very difficult to modify), and then, design a flexible, yet robust, options system using the Strategy pattern, that lets us to code as complex options as we want without modify the code base.

**Comment by Gymnasiast** [comment]
OpenTTD's interface does indeed handle lots of options while still being navigable. Still, I think OpenTTD contains lots of options that aren't really worth changing, which add nothing.

Now, my main gripe with too many options is the added complexity - in the interface, in the code and in the number of configurations we have to support. Especially because it's easy to break an option that, for example, only 1% of the user base uses. Or troubleshooting error that only pop up when using a combination of settings. And we have limited manpower. Also, we're not even finished implementing everything and we have had so many options added already. At this rate, we might even overtake OpenTTD before 1.0.0.

So far, I have only really _objected_ to a handful of options which I found ridiculously trivial. In other cases, I just commented that I didn't consider it important enough to warrant an option. This has never prevented an option from being implemented, as I have been overridden every time. I can't say I'm happy with that.

I already find this feature trivial (how long does it really take to chop off the '1'?), but having an option really takes the piss in my opinion.
